pkg/plugins/golang/v2/scaffolds/api.go

```diff
@@ -47,40 +47,31 @@ type apiScaffolder struct {
 	resource    *resource.Resource
 	// plugins is the list of plugins we should allow to transform our generated scaffolding
 	plugins []model.Plugin
-	// doResource indicates whether to scaffold API Resource or not
-	doResource bool
-	// doController indicates whether to scaffold controller files or not
-	doController bool
+	// doDefinition indicates whether to scaffold API resource definition or not
+	doDefinition bool
 }
 
 // NewAPIScaffolder returns a new Scaffolder for API/controller creation operations
 func NewAPIScaffolder(
 	config *config.Config,
 	boilerplate string,
 	res *resource.Resource,
-	doResource, doController bool,
+	doDefinition bool,
 	plugins []model.Plugin,
 ) cmdutil.Scaffolder {
 	return &apiScaffolder{
 		config:       config,
 		boilerplate:  boilerplate,
 		resource:     res,
 		plugins:      plugins,
-		doResource:   doResource,
-		doController: doController,
+		doDefinition: doDefinition,
 	}
 }
 
 // Scaffold implements Scaffolder
 func (s *apiScaffolder) Scaffold() error {
 	fmt.Println("Writing scaffold for you to edit...")
-
-	switch {
-	case s.config.IsV2(), s.config.IsV3():
-		return s.scaffold()
-	default:
-		return fmt.Errorf("unknown project version %v", s.config.Version)
-	}
```

Comment:
why we are removing the check?

Reply:
This check is redundant because the `go.kubebuilder.io/v2` plugin already specified the supported project versions. So, only those two versions can have reached this point.

Reply:
Have we the redundant check in other places? If yes, I think would be better we raise a PR to remove all from across the project instead of it.

Reply:
I think the others were removed in other PRs and this is the last one. Will double check it just in case.

Reply:
Found another one, removing it too so that they are all done.